tests/python/pants_test/engine/test_fs.py

```diff
@@ -598,6 +599,44 @@ def test_caches_downloads(self):
           82
         ))
 
+  def generate_original_digest(self) -> Digest:
+    content = b'dummy content'
+    input_files_content = InputFilesContent((
+      FileContent(path='a.txt', content=content),
+      FileContent(path='b.txt', content=content),
+      FileContent(path='c.txt', content=content),
+      FileContent(path='subdir/a.txt', content=content),
+      FileContent(path='subdir/b.txt', content=content),
+      FileContent(path='subdir2/a.txt', content=content),
+      FileContent(path='subdir2/nested_subdir/x.txt', content=content),
+    ))
+    digest = self.request_single_product(Digest, input_files_content)
+    return digest
+
+  def test_empty_snapshot_subset(self) -> None:
+    ss = SnapshotSubset(directory_digest=self.generate_original_digest(),
+        includes = PathGlobs((), ()),
+    )
+    subset_snapshot = self.request_single_product(Snapshot, ss)
+    assert subset_snapshot.directory_digest == EMPTY_DIRECTORY_DIGEST
+    assert subset_snapshot.files == ()
+    assert subset_snapshot.dirs == ()
+
+  def test_snapshot_subset_files_dirs(self) -> None:
```

Comment:
A test should be added here that validates that the digest produced by capturing some files matches the digest produced by subsetting another snapshot to contain the same content: that would have caught the bug above.

Reply:
Added this test